appium-dotnet-driver/Appium/AppiumCommandExecutionHelper.cs

```diff
@@ -58,6 +65,77 @@ public static void HideKeyboard(IExecuteMethod executeMethod, string strategy =
+                PrepareArgument("contentType", clipboardContentType.ToString().ToLower())).Value;
+        }
+
+        public static string GetClipboardText(IExecuteMethod executeMethod)
+        {
+            var encodedContentBytes = Convert.FromBase64String(GetClipboard(executeMethod, ClipboardContentType.PlainText));
+            return Encoding.UTF8.GetString(encodedContentBytes);
+        }
+
+        public static string SetClipboard(IExecuteMethod executeMethod, ClipboardContentType clipboardContentType, string base64Content)
+        {
+            switch (clipboardContentType)
+            {
+                case ClipboardContentType.Url:
+                case ClipboardContentType.PlainText:
+                    var encodedContentBytes = Encoding.UTF8.GetBytes(base64Content);
+                    base64Content = Convert.ToBase64String(encodedContentBytes);
+                    break;
+            }
+            return (string)executeMethod.Execute(AppiumDriverCommand.SetClipboard,
+                PrepareArguments(new[] { "content", "contentType", "label" },
+                    new object[] { base64Content, clipboardContentType.ToString().ToLower(), null })).Value;
+        }
+
+        public static string SetClipboardText(IExecuteMethod executeMethod, ClipboardContentType clipboardContentType, string textContent, string label)
+        {
+            if (textContent == null) throw new NullReferenceException(nameof(textContent));
+            var encodedStringContentBytes = Encoding.UTF8.GetBytes(textContent);
+
+            return (string)executeMethod.Execute(AppiumDriverCommand.SetClipboard,
+                PrepareArguments(new[] { "content", "contentType", "label" },
+                    new object[] { Convert.ToBase64String(encodedStringContentBytes), clipboardContentType.ToString().ToLower(), label })).Value;
+        }
+
+        #endregion Device Commands
+
+        /// <summary>
+        /// Prepares single command argument
+        /// </summary>
+        /// <param name="name">The parameter name</param>
+        /// <param name="value">The parameter value</param>
+        /// <returns></returns>
+        internal static Dictionary<string, object> PrepareArgument(string name, object value)
+        {
+            var parameterBuilder = new Dictionary<string, object> { { name, value } };
+            return parameterBuilder;
+        }
+
+        /// <summary>
+        /// Prepares a collection of command arguments
+        /// </summary>
+        /// <param name="names">The array of parameter names</param>
+        /// <param name="values">The array of parameter values</param>
+        /// <returns></returns>
+        internal static Dictionary<string, object> PrepareArguments(string[] names, object[] values)
+        {
+            var parameterBuilder = new Dictionary<string, object>();
+            for (var i = 0; i < names.Length; i++)
+            {
+                if (names[i] != string.Empty && values[i] != null)
```

Comment:
shouldn't null values be transformed to JSON null instead?

Reply:
No. That serialisation is handled by Selenium, and works very well